### PR TITLE
Add a few more notes to the style guide

### PR DESCRIPTION
**Pull request by Ericson2314**
- Mention 2 space indent seems to be the most ideomatic

- Mention trailing commas are good

- Update example to use trailing commas

### CODE REVIEW

File: docs/markdown/Style-guide.md [hunk 1]
```diff
@@ -9,7 +9,11 @@ Meson build files.
 
 ## Tabs or spaces?
 
-Always spaces.
+Always spaces. Two spaces is most common.
```
Comment (eli-schwartz) [inline docs/markdown/Style-guide.md:12]: Is it really?

I religiously avoid them, personally.

File: docs/markdown/Style-guide.md [hunk 2]
```diff
@@ -9,7 +9,11 @@ Meson build files.
...
+
+## Trailing commas?
+
+Yes where possible. They help make nicer diffs in version control systems.
```
Comment (eli-schwartz) [inline docs/markdown/Style-guide.md:16]: +1, the "JSON Problem" should be avoided wherever possible.

File: docs/markdown/Style-guide.md (hunk 1 above)
Comment (eli-schwartz) [reply docs/markdown/Style-guide.md:12]: Anecdotally, `muon fmt` which is supported by the vscode extension, defaults to 4 spaces but allows you to configure an arbitrary "indentation string".

Comment (dcbaker) [reply docs/markdown/Style-guide.md:12]: Most of our documentation uses two space indent, and that's what I use, so anything I've worked on has two space indent.

File: docs/markdown/Style-guide.md [hunk 3]
```diff
@@ -82,6 +86,6 @@ sources = files([
   'f1.c',
   'f2.c',
   'f10.c',
-  'f20.c'
+  'f20.c',
 ])
```
Comment (Ericson2314) [inline docs/markdown/Style-guide.md:90]:
```suggestion
  'f20.c',
)
```
In https://github.com/NixOS/nix/pull/10378#discussion_r1558205104 @eli-schwartz says he would prefer no array here

**Comment by QuLogic** [comment]
> * Mention 2 space indent seems to be the most ideomatic

s/ideomatic/idiomatic/ in the commit message.

**Comment by Ericson2314** [comment]
Fixed the commit message and removed the `[...]` in the example. Is there anything else that needs to be done for this?

**Comment by Ericson2314** [comment]
Thanks!

**Pull request merged**